dev-db/etcd/etcd-3.5.9.ebuild

```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit go-module systemd tmpfiles
+GIT_COMMIT=a603c0798
+
+DESCRIPTION="Highly-available key value store for shared configuration and service discovery"
+HOMEPAGE="https://github.com/etcd-io/etcd"
+SRC_URI="https://github.com/etcd-io/etcd/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI+=" etcd-3.5.9-deps.tar.xz"
+
+LICENSE="Apache-2.0 BSD BSD-2 MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~riscv"
+IUSE="doc +server"
+
+COMMON_DEPEND="server? (
+	acct-group/etcd
+	acct-user/etcd
+	)"
+DEPEND="${COMMON_DEPEND}"
+RDEPEND="${COMMON_DEPEND}"
+
+# Tests fail with this error:
+# fatal error: checkptr: unsafe pointer conversion
+RESTRICT="test"
+
+PATCHES=(
+	"${FILESDIR}"/etcd-3.5.9-no-gobin.patch
+	"${FILESDIR}"/etcd-3.5.9-respect-go-env-vars.patch
+)
+
+src_prepare() {
+	export GO_BUILD_FLAGS="-v -x"
+	default
+	sed -e "s|GIT_SHA=.*|GIT_SHA=${GIT_COMMIT}|"\
+		-i "${S}"/build || die
+	sed -e 's:\(for p in \)shellcheck :\1 :' \
+		-e 's:^			gofmt \\$:\\:' \
+		-e 's:^			govet \\$:\\:' \
+		-e 's:^			govet_shadow \\$:\\:' \
+		-i "${S}"/test || die
+
+	sed -e "s|GO_BUILD_FLAGS=\"[^\"]*\"|GO_BUILD_FLAGS=\"${GO_BUILD_FLAGS}\"|" \
+		-e "s|go test |go test ${GO_BUILD_FLAGS} |" \
+		-i ./test || die
+}
+
+src_compile() {
+	./build.sh || die
+	#ego build
+}
+
+src_test() {
+	./test || die
+}
+
+src_install() {
+	ego install
```

Comment:
I was just testing with etcd-3.5.10, and it seems this `ego install` actually doesn't install anything.